docs/src/main/asciidoc/kafka.adoc

```diff
@@ -3,438 +3,946 @@ This guide is maintained in the main Quarkus repository
+import org.apache.kafka.clients.consumer.TopicPartition;
 
-== The price converter
+import javax.enterprise.context.ApplicationScoped;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
 
-The price converter reads the prices from Kafka, and transforms them.
-Create the `src/main/java/org/acme/kafka/PriceConverter.java` file with the following content:
+@ApplicationScoped
+@Identifier("rebalanced-example.rebalancer")
+public class KafkaRebalancedConsumerRebalanceListener implements KafkaConsumerRebalanceListener {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaRebalancedConsumerRebalanceListener.class.getName());
+
+    /**
+     * When receiving a list of partitions will search for the earliest offset within 10 minutes
+     * and seek the consumer to it.
+     *
+     * @param consumer   underlying consumer
+     * @param partitions set of assigned topic partitions
+     */
+    @Override
+    public void onPartitionsAssigned(Consumer<?, ?> consumer, Collection<TopicPartition> partitions) {
+        long now = System.currentTimeMillis();
+        long shouldStartAt = now - 600_000L; //10 minute ago
+
+        Map<TopicPartition, Long> request = new HashMap<>();
+        for (TopicPartition partition : partitions) {
+            LOGGER.info("Assigned " + partition);
+            request.put(partition, shouldStartAt);
+        }
+        Map<TopicPartition, OffsetAndTimestamp> offsets = consumer.offsetsForTimes(request);
+        for (Map.Entry<TopicPartition, OffsetAndTimestamp> position : offsets.entrySet()) {
+            long target = position.getValue() == null ? 0L : position.getValue().offset();
+            LOGGER.info("Seeking position " + target + " for " + position.getKey());
+            consumer.seek(position.getKey(), target);
+        }
+    }
+
+}
+----
 
 [source, java]
 ----
-package org.acme.kafka;
+package inbound;
 
+import io.smallrye.reactive.messaging.kafka.IncomingKafkaRecord;
+import org.eclipse.microprofile.reactive.messaging.Acknowledgment;
 import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@ApplicationScoped
+public class KafkaRebalancedConsumer {
+
+    @Incoming("rebalanced-example")
+    @Acknowledgment(Acknowledgment.Strategy.NONE)
+    public CompletionStage<Void> consume(IncomingKafkaRecord<Integer, String> message) {
+        // We don't need to ACK messages because in this example we set offset during consumer re-balance
+        return CompletableFuture.completedFuture(null);
+    }
+
+}
+----
+
+To configure the inbound connector to use the provided listener we either set the consumer rebalance listener’s identifier:
+// Me thinking that .name is not great anymore. we may want to enhance this with .idenfier.
```

Comment:
to be removed - can you create an issue for this?

Reply:
👍